### PR TITLE
remove --feature-gates=DynamicAuditing

### PR DESCRIPTION
**Pull request by deads2k**
sig-auth has decided to remove dynamic audit after 1.5 years or so in alpha.  See mailing list thread for more details here: https://groups.google.com/forum/#!topic/kubernetes-sig-auth/aV_nXpa5uWU

/kind cleanup
/priority important-soon
@kubernetes/sig-auth-api-reviews 

```release-note
The alpha `DynamicAuditing` feature gate and `auditregistration.k8s.io/v1alpha1` API have been removed and are no longer supported.
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/listers/auditregistration/v1alpha1/BUILD [hunk 1]
```diff
@@ -10,10 +10,10 @@ go_library(
     importpath = "k8s.io/client-go/listers/auditregistration/v1alpha1",
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/listers/auditregistration/v1alpha1/BUILD:10]: I expected the auditregistration folder to be removed

File: staging/src/k8s.io/client-go/deprecated/scheme/BUILD [hunk 2]
```diff
@@ -55,6 +54,7 @@ go_library(
         "//staging/src/k8s.io/apimachinery/pkg/runtime/schema:go_default_library",
         "//staging/src/k8s.io/apimachinery/pkg/runtime/serializer:go_default_library",
         "//staging/src/k8s.io/apimachinery/pkg/util/runtime:go_default_library",
+        "//vendor/k8s.io/api/auditregistration/v1alpha1:go_default_library",
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/deprecated/scheme/BUILD:57]: I didn't expect this to still exist

File: staging/src/k8s.io/client-go/deprecated/fake/BUILD [hunk 3]
```diff
@@ -141,6 +138,9 @@ go_library(
         "//staging/src/k8s.io/client-go/discovery:go_default_library",
         "//staging/src/k8s.io/client-go/discovery/fake:go_default_library",
         "//staging/src/k8s.io/client-go/testing:go_default_library",
+        "//vendor/k8s.io/api/auditregistration/v1alpha1:go_default_library",
+        "//vendor/k8s.io/client-go/deprecated/typed/auditregistration/v1alpha1:go_default_library",
+        "//vendor/k8s.io/client-go/deprecated/typed/auditregistration/v1alpha1/fake:go_default_library",
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/deprecated/fake/BUILD:143]: drop these?

**Review by liggitt — COMMENTED** [review]
mechanical changes lgtm, a few straggling generated files, looks like

**Comment by sftim** [comment]
Hmm; the release note from this does not show up in https://relnotes.k8s.io/?areas=audit

**Pull request merged**